File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.py
def solution(s, skip, index):
    answer = ''
    
    for c in s:
        count = 0
        
        while count < index:
            if (c := chr(ord(c) + 1)) > 'z':
                c = 'a'
            if c not in skip:
                count += 1
                
        answer += c
        
    return answer

File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.java
class Solution {
    
    public String solution(String s, String skip, int index) {
        StringBuilder answer = new StringBuilder();
        
        for (char c : s.toCharArray()) {
            int count = 0;
            char current = c;
            
            while (count < index) {
                current++;
                
                if (current > 'z') {
                    current = 'a';
                }
                
                if (skip.indexOf(current) == -1) {
                    count++;
                }
            }
            
            answer.append(current);
        }
        
        return answer.toString();
    }
}